sympy/matrices/matrices.py

```diff
@@ -2301,7 +2301,17 @@ def _handle_creation_inputs(cls, *args, **kwargs):
             It only affects the creation of a matrix from the 3
             arguments, specified as ``rows, cols, func``.
 
-            Default is ``True``
+            Default is ``True``.
+
+        sympify_callback_return : Boolean, optional
+            If ``True``, it sympifies the return from the callback.
+
+            If ``False``, it skips the process.
+
+            This flag is unsafe for use if the input is not guaranteed
```

Comment:
It's the output that has to be guaranteed to be SymPy sanitized, not the input:

perhaps this, instead:

Setting this to False can safely be done if one is not returning int values from 
the callback functon as in `lambda i, j: int(i + j)` which would produce a Python int
instead of `lambda i, j: i + j` which would produce a SymPy Integer.